azurelinuxagent/common/event.py

```diff
@@ -548,11 +544,26 @@ def add_metric(self, category, counter, instance, value, log_event=False):
         event.parameters.append(TelemetryEventParam(GuestAgentPerfCounterEventsSchema.Value, float(value)))
         self.add_common_event_parameters(event, datetime.utcnow())
 
-        data = get_properties(event)
-        try:
-            self.save_event(json.dumps(data))
-        except EventError as e:
-            logger.periodic_error(logger.EVERY_FIFTEEN_MINUTES, "[PERIODIC] {0}".format(ustr(e)))
+        self.flush_or_save_event(event, message, immediate_flush)
+
+    def flush_or_save_event(self, event, message, immediate_flush):
+        """
+        Flush the event to wireserver if immediate_flush to set to true, else
+        save it disk if we fail to send or not required to flush immediately.
+        """
+        report_success = immediate_flush
+        if immediate_flush and self.protocol is not None:
```

Comment:
if self.protocol is None and immediate_flush is True then we never report the event. Is this intentional?